### PR TITLE
Add sentry-crashpad/0.4.17

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **sentry-crashpad/0.4.17**

I also clean old versions including 0.4.14 since sentry-native doesn't handle it.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
Hi @jgsogo ! I wondered why the recipe take so long to finish. Shall I close and reopen the PR again ?

**Comment by jgsogo** [comment]
I've triggered it again manually.

File: recipes/sentry-crashpad/all/conandata.yml [hunk 1]
```diff
@@ -1,67 +1,17 @@
 sources:
+  "0.4.17":
+    url: "https://github.com/getsentry/sentry-native/releases/download/0.4.17/sentry-native.zip"
+    sha256: "38c9e180c29b6561c5f91ada154e191eabe9d7c270734cff81e55532878ec273"
   "0.4.15":
     url: "https://github.com/getsentry/sentry-native/releases/download/0.4.15/sentry-native.zip"
     sha256: "ae3ac4efa76d431d8734d7b0b1bf9bbedaf2cbdb18dfc5c95e2411a67808cf29"
-  "0.4.14":
```
Comment (uilianries) [inline recipes/sentry-crashpad/all/conandata.yml:5]: Why did you remove 14 but kept 13?

Comment (MartinDelille) [reply recipes/sentry-crashpad/all/conandata.yml:5]: Because this recipe is a dependency of sentry-native which doesn't have 0.4.14 version: https://github.com/conan-io/conan-center-index/blob/master/recipes/sentry-native/config.yml

Comment (uilianries) [reply recipes/sentry-crashpad/all/conandata.yml:5]: Good catch!

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/sentry-crashpad/all/conanfile.py [hunk 2]
```diff
@@ -40,8 +40,8 @@ def _source_subfolder(self):
     @property
     def _minimum_compilers_version(self):
         return {
-            "Visual Studio": "15",
-            "gcc": "5",
+            "Visual Studio": "15" if tools.Version(self.version) < "0.4.16" else "16",
+            "gcc": "9",
```
Comment (ericriff) [inline recipes/sentry-crashpad/all/conanfile.py:44]: This broke my builds (we use gcc 8).
Why do you think we need gcc 9 to build with C++14? These docs say 5 is enough.
https://gcc.gnu.org/projects/cxx-status.html#cxx14

Comment (uilianries) [reply recipes/sentry-crashpad/all/conanfile.py:44]: Depends the feature ... for this case, it seems like gcc 5 has what we need. We can drop to 8, for sure.
